src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -343,3 +346,165 @@ class Config:
     @accepts(Int('id'))
     async def do_delete(self, id):
         return await self.middleware.call('datastore.delete', self._config.datastore, id)
+
+
+class RsyncTaskService(CRUDService):
+
+    class Config:
+        datastore = 'tasks.rsync'
+        datastore_prefix = 'rsync_'
+
+    async def validate_rsync_task(self, data):
+        verrors = ValidationErrors()
+
```

Comment:
Lets not lose comments gratuitously when moving logic from GUI, please. They might be helpful in the future.

```
-        # Windows users can have spaces in their usernames
-        # http://www.freebsd.org/cgi/query-pr.cgi?pr=164808
```